server/src/main/java/org/elasticsearch/node/NodeService.java

```diff
@@ -133,6 +139,20 @@ public NodeInfo info(
         );
     }
 
+    private Map<String, Integer> findComponentVersions() {
+        var versions = pluginService.loadServiceProviders(ComponentVersionNumber.class)
+            .stream()
+            .collect(Collectors.toUnmodifiableMap(ComponentVersionNumber::componentId, cvn -> cvn.versionNumber().id()));
+
+        if (Assertions.ENABLED) {
+            var hasUpperCase = Pattern.compile("\\p{Upper}").asPredicate();
```

Comment:
If we are going to enforce snake case, then shouldn't we invert this and check it only contains lowercase ascii and underscores? For example, this does not prohibit dashes, dots, etc.

Reply:
Yeah, good idea